Fix/Refactor the model XMLs

lukecologne opened the pull request:
<!-- ⚠⚠ Do not delete this pull request template! ⚠⚠ -->
<!-- Pull requests that do not follow this template are likely to be ignored. -->

## Summary of Changes
<!-- Please provide a summary of changes for this pull request, ensuring all changes are explained. -->
- [x] Add additional features to the FBW_(Covered)_Push_Toggle
- [x] Change mutliple buttons to FBW_Push_Toggle, and remove the associated, very specific, templates
       (This also fixes missing sounds with those buttons)
- [x] Add FBW_Push_Held_NoEmissive generic template, convert buttons to this template (call buttons, fire test)
   - [ ] Convert ND chrono button to this template
- [x] Refactor the engine fire buttons
   - [x] engine fire guards can be closed now, AGENT buttons are now animated when pressed
- [x] Seperate custom generic templates into their own file for clarity 
- [x] Move all edited templates from the Asobo/Common folder to our own XML files (Asobo/Airliner/Airbus should be fine to leave as is)
- [x] APU STARTER button tooltip now doesn't always show (unavailable)


(This is mostly a under-the-hood change)

<!-- DO NOT DELETE THIS -->
## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build.py will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, slick on the **Artifacts** drop down and click the **A32NX** link

Benjozork commented on A32NX/ModelBehaviorDefs/A32NX/A32NX_Interior.xml:
```diff
@@ -543,4 +576,138 @@
+            <SEQ2_EMISSIVE_CODE>(L:FIRE_TEST_#ID#, Bool) (L:XMLVAR_SWITCH_OVHD_INTLT_ANNLT_Position) 0 == or (L:XMLVAR_#NODE_ID#_Discharge) or</SEQ2_EMISSIVE_CODE>
+            <LEFT_SINGLE_CODE>
+                (L:FIRE_BUTTON_#ID#, Bool) if{
+                1 (>L:XMLVAR_#NODE_ID#_Discharge)
```
```suggestion
                    1 (>L:XMLVAR_#NODE_ID#_Discharge)
```

Benjozork commented on A32NX/ModelBehaviorDefs/A32NX/A32NX_Interior.xml:
```diff
@@ -543,4 +576,138 @@
+        <Component ID="#NODE_ID#" Node="#NODE_ID#">
+
+            <Update Frequency = "10">
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 0 &gt; (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#_IsDown) 0 &gt; or if{
+                (O:AnimGoingForward) if{
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 0.1 #ANIM_DURATION# / + 1 min (&gt;#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#)
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 1 == if{
+                0 (&gt;O:AnimGoingForward)
+                }
+                } els{
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 0.1 #ANIM_DURATION# / - (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#_IsDown) max (&gt;#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#)
+                }
+                }
```
indentation

Benjozork commented on A32NX/ModelBehaviorDefs/A32NX/A32NX_Interior.xml:
```diff
@@ -543,4 +576,138 @@
+            <UseTemplate Name = "ASOBO_GT_Interaction_LeftSingle_Leave_Code">
+                <LEFT_SINGLE_CODE>
+                    (L:FIRE_GUARD_#ID#) 1 == if{
+                    1 (&gt;#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#_IsDown) 1 (&gt;O:AnimGoingForward) #LEFT_SINGLE_CODE#
```
```suggestion
                        1 (&gt;#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#_IsDown) 1 (&gt;O:AnimGoingForward) #LEFT_SINGLE_CODE#
```

Benjozork commented on A32NX/ModelBehaviorDefs/A32NX/A32NX_Interior.xml:
```diff
@@ -543,4 +576,138 @@
+        <Component ID="#NODE_ID#" Node="#NODE_ID#">
+
+            <Update Frequency = "10">
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 0 &gt; (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#_IsDown) 0 &gt; or if{
+                (O:AnimGoingForward) if{
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 0.1 #ANIM_DURATION# / + 1 min (&gt;#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#)
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 1 == if{
+                0 (&gt;O:AnimGoingForward)
+                }
+                } els{
+                (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#) 0.1 #ANIM_DURATION# / - (#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#_IsDown) max (&gt;#SWITCH_POSITION_TYPE#:#SWITCH_POSITION_VAR#)
+                }
+                }
```
indentation

Benjozork commented:
> FBW_Push_Held_NoEmissive

Why is a special NoEmissive variant needed ?

lukecologne commented:
I mean it's not a special variant, it's a new template. I just didn't add an emissive to it.

lukecologne commented:
Not all the shared XML files are removed yet, but all the ones from the Asobo/Common folder are, and I think that is enough for one PR. The files that remain are Asobo/Airliner/(Airbus, AirlinerCommon and FMC). Airbus.xml should be fine for now, since it is exclusive to the A320, but the others are shared files as well and should be removed.

Benjozork commented:
LGTM, I think we should get QA to test if anything regressed.

lukecologne commented:
To whoever merges this: please, please use squash merge, I don't want all those 18 commits in the history

lukecologne commented:
Question: Do you think I should further seperate the templates into different files? Like electrical/fire/etc. so it doesn't get too crowded in this single file when we transfer more templates over

Benjozork commented:
> 
> 
> Question: Do you think I should further seperate the templates into different files? Like electrical/fire/etc. so it doesn't get too crowded in this single file when we transfer more templates over

Yes.

Carter633 commented:
Name: British Avgeek YT -- Carter633 #1436
Date of testing: Current date 26/10/2020
Version of the sim: 1.9.5.0
PR Tested: #1476 
Object of test: Check all function of the A32NX for hard fork

Testing technique: Cold & Dark - Full flight on VATSIM, Heathrow to Rome
Steps you've made to test the Pull Request:
1.  Spawned at Gate, avionics on - tested all buttons, dials, lights etc for correct function
2. Full flight plan input in MCDU including airways page and some DCT to waypoints, also selecting SIDs and STARs and using Plan mode on ND to verify it all
3. APU On, rechecked all buttons, dials etc for any changes or issues
4. Pushback and Start - rechecked everything as above
5. During flight - rechecked everything as above

Flight notes

Startup: Everything is normal
Taxi/Takeoff : Everything is normal
Cruise: Everything is normal
Descent/Landing : Work as expected
Shutdown: As expected - worked as normal

Issues (If Any): None found
Overall Rating: Good
Conclusions of the testing: Checked all functions possible and everything seemed to work as it should and as it did prior to it all being pulled to another file set

lukecologne commented:
Alright, I have moved the templates into seperate files, and also moved them together in a new subfolder. Some templates are still in a misc file, I'll create appropriate files for those once we move more templates to our own files.

Carter633 commented:
Name: British Avgeek YT -- Carter633 #1436
Date of testing: Current date 26/10/2020
Version of the sim: 1.9.5.0
PR Tested: #1476
Object of test: Check all function of the A32NX for hard fork (test 2)

Testing technique: Cold & Dark - Shorter flight from Heathrow to Manchester
Steps you've made to test the Pull Request:

1. This time used flight planning in the world map prior to spawning to check any other issues that may arise and to run the test in a different set of conditions
2. Avionics on, checked all buttons, alarms etc, APU On, rechecked all buttons, dials etc for any changes or issues
3. Pushback and Start - rechecked everything as above
4. During flight - rechecked everything as above

Flight notes
Startup: Everything is normal
Taxi/Takeoff : Everything is normal
Cruise: Everything is normal
Descent/Landing : Work as expected
Shutdown: As expected - worked as normal

Issues (If Any): None found
Overall Rating: Good
Conclusions of the testing: Checked all functions possible and everything seemed to work as it should and as it did prior to it all being pulled to another file set

starvingjavi commented:
Name: Javi#2557
Date of testing: Current date 26/10/2020
Version of the sim: 1.9.5.0
PR Tested: #1476
Object of test: Check all functions of the A32NX for hard fork on normal flight

Testing technique: Short flight from MMMX to MMAA started from C&D
Steps you've made to test the Pull Request:

Used flight plan from SimBrief and loaded it through the world map
Checked all overhead buttons and no problems were detected.
Pushback and Start - Followed checklist
During flight - Followed Checklist and descent checklist
Flight notes
Startup: Normal
Taxi/Takeoff : Normal
Cruise: Normal
Descent/Landing : As expected
Shutdown: Normal

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: Checked all buttons that would be used on a normal flight plus some additional ones to check for bugs and errors.
